lib/matplotlib/figure.py

```diff
@@ -1242,6 +1242,20 @@ def colorbar(
             fig.sca(current_ax)
             cax.grid(visible=False, which='both', axis='both')
 
+        if hasattr(mappable, "figure") and mappable.figure is not None:
+            # Get top level artists
+            mappable_host_fig = mappable.figure
+            self_host_fig = self.figure
```

Comment:
I would not declare an extra variable for this. It’s only used once and using self.figure is fine there.

Reply:
Will fix this.